src/aws-cpp-sdk-core/include/smithy/interceptor/impl/ChecksumInterceptor.h

```diff
@@ -0,0 +1,209 @@
+                    }
+                    else if (headers.find(Aws::Http::CONTENT_MD5_HEADER) == headers.end())
+                    {
+                        AWS_LOGSTREAM_WARN(CHECKSUM_INTERCEPTOR_LOG_TAG, "Checksum algorithm: " << checksumAlgorithmName << " is not supported by SDK.");
+                    }
+                }
+
+                // Response checksums
+                if (modeledRequest.ShouldValidateResponseChecksum())
+                {
+                    for (const Aws::String& responseChecksumAlgorithmName : modeledRequest.GetResponseChecksumAlgorithmNames())
+                    {
+                        const auto lowered = Aws::Utils::StringUtils::ToLower(responseChecksumAlgorithmName.c_str());
+
+                        if (lowered == "crc32c")
+                        {
+                            std::shared_ptr<Aws::Utils::Crypto::CRC32C> crc32c = Aws::MakeShared<Aws::Utils::Crypto::CRC32C>(CHECKSUM_INTERCEPTOR_LOG_TAG);
+                            httpRequest->AddResponseValidationHash("crc32c", crc32c);
+                        }
+                        else if (lowered == "crc32")
+                        {
+                            std::shared_ptr<Aws::Utils::Crypto::CRC32> crc32 = Aws::MakeShared<Aws::Utils::Crypto::CRC32>(CHECKSUM_INTERCEPTOR_LOG_TAG);
+                            httpRequest->AddResponseValidationHash("crc32", crc32);
+                        }
+                        else if (lowered == "sha1")
+                        {
+                            std::shared_ptr<Aws::Utils::Crypto::Sha1> sha1 = Aws::MakeShared<Aws::Utils::Crypto::Sha1>(CHECKSUM_INTERCEPTOR_LOG_TAG);
+                            httpRequest->AddResponseValidationHash("sha1", sha1);
+                        }
+                        else if (lowered == "sha256")
+                        {
+                            std::shared_ptr<Aws::Utils::Crypto::Sha256> sha256 = Aws::MakeShared<Aws::Utils::Crypto::Sha256>(CHECKSUM_INTERCEPTOR_LOG_TAG);
+                            httpRequest->AddResponseValidationHash("sha256", sha256);
+                        }
+                        else
+                        {
+                            AWS_LOGSTREAM_WARN(CHECKSUM_INTERCEPTOR_LOG_TAG, "Checksum algorithm: " << lowered << " is not supported in validating response body yet.");
+                        }
+                    }
+                }
+                return context.GetTransmitRequest();
+            }
+
+            ModifyResponseOutcome ModifyBeforeDeserialization(InterceptorContext& context) override
+            {
+                const auto& request = context.GetModeledRequest();
+                const auto transmitRequest = context.GetTransmitRequest();
+                const auto response = context.GetTransmitResponse();
+                if (transmitRequest == nullptr || transmitRequest == nullptr)
```

Comment:
nit: condense if (!(transmitRequest )
Duplicate : if (transmitRequest == nullptr || transmitRequest == nullptr), you meant response

Reply:
> you meant response

yep good catch


> nit: condense if (!(transmitRequest ))

[this is the code style that crt uses](https://github.com/search?q=repo%3Aawslabs%2Faws-crt-cpp%20%22%3D%3D%20nullptr%22&type=code) so i wanted to keep it consistent